src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -65,28 +58,39 @@ public NsUser getWriter() {
 
     public void addAnswer(Answer answer) {
         answer.toQuestion(this);
+        //List<Answer> newAnswers = new ArrayList<>();
+        //newAnswers.add(answer);
+        if(answers == null) {
+            answers = new Answers(new ArrayList<>());
+        }
         answers.add(answer);
     }
 
     public boolean isOwner(NsUser loginUser) {
         return writer.equals(loginUser);
     }
 
-    public Question setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
-    }
-
     public boolean isDeleted() {
         return deleted;
     }
 
     public List<Answer> getAnswers() {
-        return answers;
+        return answers.getAnswers();
+    }
+
+    public void deleteQuestion(NsUser loginUser) throws CannotDeleteException {
+        isOwnerCheck(loginUser);
+        this.deleted = true;
     }
 
     @Override
     public String toString() {
-        return "Question [id=" + getId() + ", title=" + title + ", contents=" + contents + ", writer=" + writer + "]";
+        return "Question [id=" + getId() + ", title=" + questionBoard.getTitle() + ", contents=" + questionBoard.getContents() + ", writer=" + writer + "]";
+    }
+
+    public void isOwnerCheck(NsUser loginUser) throws CannotDeleteException {
```

Comment:
```suggestion
    public void checkOwner(NsUser loginUser) throws CannotDeleteException {
```
isXXX 네이밍은 boolean 을 반환하는 것으로 오해할 수 있어보여요
validate나 check 같은 표현이 어울릴 것 같습니다.